src/Utilities/FileSystem.hpp

```diff
@@ -0,0 +1,139 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Declares functions to do file system manipulations
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+/*!
+ * \ingroup FileSystem
+ * \brief A light-weight file system library based on POSIX.
+ *
+ * We use this library instead of a subprocess based library because OpenMPI
+ * does not support forking of processes on all systems. Since the
+ * parallelization library we use may be implemented on top of OpenMPI we
+ * take the safe route and use POSIX.
+ */
+namespace filesystem {
+
+/*!
+ * \ingroup FileSystem
+ * \brief Wraps the dirname function to get the pathname of the parent directory
+ *
+ * See the opengroup documentation:
+ * http://pubs.opengroup.org/onlinepubs/009695399/functions/dirname.html
+ *
+ * \example
+ * \snippet Test_FileSystem.cpp dirname
+ *
+ * \requires `file_path` is a valid path on the filesystem
+ * \returns the path to the file
+ */
+std::string dirname(const std::string& path);
```

Comment:
Something like `get_parent_path` seems more inline with our usual naming.  Let's do it.